In [26]:
import pandas as pd
data1 = pd.read_csv('data_cleaned_V2.csv',engine='python', dtype=str)
data2 = pd.read_csv('data_perspective_api_scores.csv',engine='python', dtype=str)

In [31]:
from sklearn.metrics import accuracy_score,classification_report,confusion_matrix

In [28]:
data1['TOXICITY'] = data2['TOXICITY'].apply(lambda score: '1' if float(score) >=0.5 else '0')

In [35]:
data1['SEVERE_TOXICITY'] = data2['SEVERE_TOXICITY'].apply(lambda score: '1' if float(score) >=0.5 else '0')

In [27]:
data1['isHate'].value_counts()

0    335
1    119
Name: isHate, dtype: int64

In [29]:
data1['TOXICITY'].value_counts()

0    263
1    191
Name: TOXICITY, dtype: int64

In [36]:
data1['SEVERE_TOXICITY'].value_counts()

0    322
1    132
Name: SEVERE_TOXICITY, dtype: int64

In [38]:
data1.head()

Unnamed: 0  isHate                                            comment  \
0          0     0.0                             Cars cars cars cars...   
1          1     0.0  If they hide in the engine they would get kill...   
2          2     0.0                   Build that wall deport them all!   
3          3     0.0  Jon Snow - The King in the north thank you for...   
4          4     0.0  Stopping illegal immigration should have been ...   

   TOXICITY SEVERE_TOXICITY  
0       0.0               0  
1       1.0               0  
2       1.0               1  
3       0.0               0  
4       0.0               0

In [34]:
accuracy_score(data1['isHate'],data1['TOXICITY'])

0.748898678414097

In [40]:
accuracy_score(data1['isHate'],data1['SEVERE_TOXICITY'])

0.8039647577092511

In [32]:
data1['TOXICITY'] = data1['TOXICITY'].astype(float)

In [39]:
data1['SEVERE_TOXICITY'] = data1['SEVERE_TOXICITY'].astype(float)

In [33]:
data1['isHate'] = data1['isHate'].astype(float)

### Out of 119 Hate comments, 98 are classified as Toxic by perspective and 21 are not Toxic

In [68]:
data1[data1['isHate']==1]['TOXICITY'].value_counts()

1.0    98
0.0    21
Name: TOXICITY, dtype: int64

#### For Severe Toxicity

In [64]:
data1[data1['isHate']==1]['SEVERE_TOXICITY'].value_counts()

1.0    81
0.0    38
Name: SEVERE_TOXICITY, dtype: int64

### Out of 335 Non-Hate comments, 93 are classified as Toxic by perspective and 242 are not Toxic

In [61]:
data1[data1['isHate']==0]['TOXICITY'].value_counts()

0.0    242
1.0     93
Name: TOXICITY, dtype: int64

#### For Severe Toxicity

In [63]:
data1[data1['isHate']==0]['SEVERE_TOXICITY'].value_counts()

0.0    284
1.0     51
Name: SEVERE_TOXICITY, dtype: int64

In [70]:
data1['IDENTITY_ATTACK'] = data2['IDENTITY_ATTACK'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['INSULT'] = data2['INSULT'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['PROFANITY'] = data2['PROFANITY'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['THREAT'] = data2['THREAT'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['SEXUALLY_EXPLICIT'] = data2['SEXUALLY_EXPLICIT'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['FLIRTATION'] = data2['SEVERE_TOXICITY'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['ATTACK_ON_AUTHOR'] = data2['ATTACK_ON_AUTHOR'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['ATTACK_ON_COMMENTER'] = data2['ATTACK_ON_COMMENTER'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['INCOHERENT'] = data2['INCOHERENT'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['INFLAMMATORY'] = data2['INFLAMMATORY'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['LIKELY_TO_REJECT'] = data2['LIKELY_TO_REJECT'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['OBSCENE'] = data2['OBSCENE'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['SPAM'] = data2['SPAM'].apply(lambda score: '1' if float(score) >=0.5 else '0')
data1['UNSUBSTANTIAL'] = data2['UNSUBSTANTIAL'].apply(lambda score: '1' if float(score) >=0.5 else '0')

In [80]:
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['IDENTITY_ATTACK'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['INSULT'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['PROFANITY'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['THREAT'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['SEXUALLY_EXPLICIT'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['FLIRTATION'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['ATTACK_ON_AUTHOR'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['ATTACK_ON_COMMENTER'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['INCOHERENT'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['INFLAMMATORY'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['LIKELY_TO_REJECT'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['OBSCENE'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['SPAM'].value_counts(),'\n')
print(data1[data1['isHate']==0][data1['TOXICITY']==1]['UNSUBSTANTIAL'].value_counts(),'\n')

1    64
0    29
Name: IDENTITY_ATTACK, dtype: int64 

1    76
0    17
Name: INSULT, dtype: int64 

1    59
0    34
Name: PROFANITY, dtype: int64 

0    66
1    27
Name: THREAT, dtype: int64 

0    72
1    21
Name: SEXUALLY_EXPLICIT, dtype: int64 

1    51
0    42
Name: FLIRTATION, dtype: int64 

0    87
1     6
Name: ATTACK_ON_AUTHOR, dtype: int64 

0    69
1    24
Name: ATTACK_ON_COMMENTER, dtype: int64 

0    62
1    31
Name: INCOHERENT, dtype: int64 

1    62
0    31
Name: INFLAMMATORY, dtype: int64 

1    89
0     4
Name: LIKELY_TO_REJECT, dtype: int64 

0    47
1    46
Name: OBSCENE, dtype: int64 

0    92
1     1
Name: SPAM, dtype: int64 

0    51
1    42
Name: UNSUBSTANTIAL, dtype: int64 



C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.
C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  
C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  This is separate from the ipykernel package so we can avoid doing imports until
C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:4: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  after removing the cwd from sys.path.
C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:5: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """
C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:6: UserWarning: Boolean Series key

In [77]:
print(data1[data1['isHate']==0]['IDENTITY_ATTACK'].value_counts(),'\n')
print(data1[data1['isHate']==0]['INSULT'].value_counts(),'\n')
print(data1[data1['isHate']==0]['PROFANITY'].value_counts(),'\n')
print(data1[data1['isHate']==0]['THREAT'].value_counts(),'\n')
print(data1[data1['isHate']==0]['SEXUALLY_EXPLICIT'].value_counts(),'\n')
print(data1[data1['isHate']==0]['FLIRTATION'].value_counts(),'\n')
print(data1[data1['isHate']==0]['ATTACK_ON_AUTHOR'].value_counts(),'\n')
print(data1[data1['isHate']==0]['ATTACK_ON_COMMENTER'].value_counts(),'\n')
print(data1[data1['isHate']==0]['INCOHERENT'].value_counts(),'\n')
print(data1[data1['isHate']==0]['INFLAMMATORY'].value_counts(),'\n')
print(data1[data1['isHate']==0]['LIKELY_TO_REJECT'].value_counts(),'\n')
print(data1[data1['isHate']==0]['OBSCENE'].value_counts(),'\n')
print(data1[data1['isHate']==0]['SPAM'].value_counts(),'\n')
print(data1[data1['isHate']==0]['UNSUBSTANTIAL'].value_counts(),'\n')

0    227
1    108
Name: IDENTITY_ATTACK, dtype: int64 

0    255
1     80
Name: INSULT, dtype: int64 

0    275
1     60
Name: PROFANITY, dtype: int64 

0    293
1     42
Name: THREAT, dtype: int64 

0    307
1     28
Name: SEXUALLY_EXPLICIT, dtype: int64 

0    284
1     51
Name: FLIRTATION, dtype: int64 

0    320
1     15
Name: ATTACK_ON_AUTHOR, dtype: int64 

0    271
1     64
Name: ATTACK_ON_COMMENTER, dtype: int64 

1    170
0    165
Name: INCOHERENT, dtype: int64 

0    185
1    150
Name: INFLAMMATORY, dtype: int64 

1    281
0     54
Name: LIKELY_TO_REJECT, dtype: int64 

0    273
1     62
Name: OBSCENE, dtype: int64 

0    322
1     13
Name: SPAM, dtype: int64 

1    230
0    105
Name: UNSUBSTANTIAL, dtype: int64 



In [78]:
print(data1[data1['isHate']==1]['IDENTITY_ATTACK'].value_counts(),'\n')
print(data1[data1['isHate']==1]['INSULT'].value_counts(),'\n')
print(data1[data1['isHate']==1]['PROFANITY'].value_counts(),'\n')
print(data1[data1['isHate']==1]['THREAT'].value_counts(),'\n')
print(data1[data1['isHate']==1]['SEXUALLY_EXPLICIT'].value_counts(),'\n')
print(data1[data1['isHate']==1]['FLIRTATION'].value_counts(),'\n')
print(data1[data1['isHate']==1]['ATTACK_ON_AUTHOR'].value_counts(),'\n')
print(data1[data1['isHate']==1]['ATTACK_ON_COMMENTER'].value_counts(),'\n')
print(data1[data1['isHate']==1]['INCOHERENT'].value_counts(),'\n')
print(data1[data1['isHate']==1]['INFLAMMATORY'].value_counts(),'\n')
print(data1[data1['isHate']==1]['LIKELY_TO_REJECT'].value_counts(),'\n')
print(data1[data1['isHate']==1]['OBSCENE'].value_counts(),'\n')
print(data1[data1['isHate']==1]['SPAM'].value_counts(),'\n')
print(data1[data1['isHate']==1]['UNSUBSTANTIAL'].value_counts(),'\n')

1    96
0    23
Name: IDENTITY_ATTACK, dtype: int64 

1    96
0    23
Name: INSULT, dtype: int64 

1    76
0    43
Name: PROFANITY, dtype: int64 

0    64
1    55
Name: THREAT, dtype: int64 

0    85
1    34
Name: SEXUALLY_EXPLICIT, dtype: int64 

1    81
0    38
Name: FLIRTATION, dtype: int64 

0    117
1      2
Name: ATTACK_ON_AUTHOR, dtype: int64 

0    105
1     14
Name: ATTACK_ON_COMMENTER, dtype: int64 

0    77
1    42
Name: INCOHERENT, dtype: int64 

1    96
0    23
Name: INFLAMMATORY, dtype: int64 

1    117
0      2
Name: LIKELY_TO_REJECT, dtype: int64 

1    62
0    57
Name: OBSCENE, dtype: int64 

0    118
1      1
Name: SPAM, dtype: int64 

0    69
1    50
Name: UNSUBSTANTIAL, dtype: int64 



### @ekarati